packages/pigeon/lib/dart_generator.dart

```diff
@@ -178,11 +178,11 @@ void _writeFlutterApi(
 String _addGenericTypes(String dataType, String nullTag) {
   switch (dataType) {
     case 'List':
-      return 'List<Object$nullTag>';
+      return 'List<Object$nullTag>$nullTag';
     case 'Map':
-      return 'Map<Object$nullTag, Object$nullTag>';
+      return 'Map<Object$nullTag, Object$nullTag>$nullTag';
     default:
-      return dataType;
+      return '$dataType$nullTag';
```

Comment:
This is the actual fix, other stuff is just testing infrastructure / tests.

Reply:
makes sense